backend/app/controllers/spree/admin/quick_switch_controller.rb

```diff
@@ -0,0 +1,94 @@
+module Spree
+  module Admin
+    class QuickSwitchController < Spree::Admin::BaseController
+      layout false
+
+      def find_object
+        quick_switch_item = Spree::Backend::Config.quick_switch_items.detect do |item|
+          item.search_terms.include? searched_key.to_sym
+        end
+
+        if quick_switch_item
+          method(quick_switch_item.method).call
+        else
+          message = Spree.t("quick_switch.invalid_query")
+          respond_to do |format|
+            format.html { redirect_back(fallback_location: spree.admin_path, flash: { error: message }) }
+            format.json { render json: { message: message }, status: :bad_request }
+          end
+        end
+      end
+
+      private
+
+      def redirect_to_url(url)
+        respond_to do |format|
+          format.html { redirect_to url }
+          format.json { render json: { redirect_url: url }, status: :ok }
+        end
+      end
+
+      def not_found(message)
+        respond_to do |format|
+          format.html { redirect_back(fallback_location: spree.admin_path, flash: { error: message }) }
+          format.json { render json: { message: message }, status: :not_found }
+        end
+      end
+
+      def searched_key
+        params[:quick_switch_query].split(" ")[0]
+      end
+
+      def searched_value
+        params[:quick_switch_query].split(" ")[1]
+      end
+
+      def find_and_redirect_to_order
```

Comment:
Now that we have a `QuickSwitchItem` class I wonder if that's where these methods should be defined.